common/dynamicconfig/file_based_client.go

```diff
@@ -166,9 +169,8 @@ func (fc *fileBasedClient) GetMapValue(
 	return defaultValue, errors.New("value type is not map")
```

Comment:
not actually relevant to this PR unless you feel like it, but:
might be useful to change this error message to include the actual type, via `%T`